app/controllers/collection_drafts_controller.rb

```diff
@@ -89,6 +59,9 @@ def create
       end
     else # record update failed
       flash[:error] = I18n.t("controllers.draft.#{plural_resource_name}.create.flash.error")
+      load_umm_schema
+      new_view_setup
+      @template_names = names_list if resource_name == 'collection_template'
```

Comment:
could this be moved into the subclass method?

Reply:
It just feels a little bit weird to be adding a call in the superclass method that is required and will only be run by the subclass.

Do the names need to be set at this particular point? Could they be set before super is called in the subclass? Or potentially with a before_action in the subclass?

Reply:
They could be called in the template's create before the super call.  I'll make the change in the morning.

Reply:
I've moved it before the super call in the latest commit.